pkg/util/k8sutil/backup.go

```diff
@@ -320,14 +320,22 @@ func CopyVolume(kubecli *unversioned.Client, fromClusterName, toClusterName, ns
 		return err
 	}
 
-	w, err := kubecli.Pods(ns).Watch(api.SingleObject(api.ObjectMeta{Name: pod.Name}))
-	if err != nil {
-		return err
-	}
-	// It could take long due to delay of k8s controller detaching the volume
-	_, err = watch.Until(120*time.Second, w, unversioned.PodCompleted)
+	// Delay could be very long due to k8s controller detaching the volume
+	err := retryutil.Retry(10*time.Second, 12, func() (bool, error) {
+		p, err := kubecli.Pods(ns).Get(pod.Name)
+		if err != nil {
+			return false, err
+		}
+		switch p.Status.Phase {
+		case api.PodSucceeded:
+			return true, nil
+		case api.PodFailed:
+			return false, errors.New("job failed")
```

Comment:
even if we are going to wrap it, we should never return a meaningless/non-descriptive error.